packages/integration-tests/src/helper.ts

```diff
@@ -55,4 +56,34 @@ export class Helper {
     }
     return ethers.utils.formatUnits(await provider.getBalance(walletAddress), "wei");
   }
+
+  async delay(ms: number) {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+  }
+
+  async performGETrequest(apiRoute: string) {
+    return request(environment.blockExplorerAPI).get(apiRoute);
+  }
+
+  /**
+   * A retry wrapper method to enhance test stability in API testing.
+   * Useful when API response fields may not immediately reflect the expected state,
+   * but can update to the correct response after a delay.
+   * Attempts to execute the action a specified number of times (defined in localConfig.maxAPIretries)
+   * with a delay between attempts (localConfig.intervalAPIretries).
+   * Throws an error if the action consistently fails after all retries.
+   */
+  async retryTestAction(action) {
+    for (let i = 0; i < localConfig.maxAPIretries; i++) {
+      try {
+        await action();
+        return;
+      } catch (error) {
+        if (i === localConfig.maxAPIretries - 1) {
+          throw error;
+        }
+        await new Promise((resolve) => setTimeout(resolve, localConfig.intervalAPIretries));
```

Comment:
Not sure, but may be here you planned to use: await delay(localConfig.intervalAPIretries)

Reply:
The solution looks elegant. Im just surprising, could we make a DDoS behavior using 200 iteration for each action() ? 
Probably you tried it, Im just interesting will we have the same effect or less DDoS in case of: 
```
maxAPIretries: 200, ---> 40
intervalAPIretries: 0.2 * 1000, ---> 1 * 1000
```
 or in some other similar proportion compensating iterations by increasing the timeout?